sdlib/d/gc/region.d

```diff
@@ -144,6 +178,7 @@ private:
 
 			r.merge(adjacent);
```

Comment:
The merging here does not keep track of dirty pages, which was the whole point of this PR.

Reply:
No. Region.merge asserts a few things as it assume they are true, when they clearly do not necessarily are, which would have been obvious in the presence of tests.

Why do I need to runt he code through my mind to figure this out? If I'm wrong, it's trivial to demonstrate with a test that checks for the various permutations of things.